### PR TITLE
added sharing a post feature

### PR DESCRIPTION
**Pull request by adrean-hipe**
### OVERVIEW

- added share method in PostController
- added service for sharing posts
- added migration to insert new column in the user_posts table name original_post_id
- added connection to the UserPost model
- modified blades to display if the post is shared
- added share button to the post actions
- added route for share method (fixed merge conflict)
- added ignore for docblocks in larastan @param

### NOTES

- php artisan migrate

### SNIPPETS

shared post
image

original post
image

### CODE REVIEW

**Review by bpocrafael — APPROVED** [review]

File: laravel-app/database/migrations/2023_10_09_030758_add_original_post_id_to_user_posts_table.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up()
```
Comment (bpocrafael) [inline laravel-app/database/migrations/2023_10_09_030758_add_original_post_id_to_user_posts_table.php:12]: add reverse

File: laravel-app/app/Services/PostService.php [hunk 2]
```diff
@@ -25,12 +23,8 @@ public function createPost(User $user, array $validatedData): UserPost
 
     /**
      * To update an existing post.
-     *
-     * @param UserPost $post
-     * @param array<string, mixed> $validatedData
-     * @return bool
      */
-    public function updatePost(UserPost $post, array $validatedData): bool
+    public function updatePost(UserPost $post, array $validatedData): bool // @phpstan-ignore-line
```
Comment (bpocrafael) [inline laravel-app/app/Services/PostService.php]: why is there phpstan-ignore-line here??

File: laravel-app/app/Services/PostService.php [hunk 3]
```diff
@@ -10,10 +10,8 @@ class PostService implements PostServiceInterface
 {
     /**
      * To create a new post.
-     * @param User $user
-     * @param array<mixed, mixed> $validatedData
      */
-    public function createPost(User $user, array $validatedData): UserPost
+    public function createPost(User $user, array $validatedData): UserPost // @phpstan-ignore-line
```
Comment (bpocrafael) [inline laravel-app/app/Services/PostService.php]: why ignore?

File: laravel-app/app/Interfaces/PostServiceInterface.php [hunk 4]
```diff
@@ -8,8 +8,17 @@
 interface PostServiceInterface
 {
     /**
-     * @param User $user
-     * @param array<mixed, mixed> $validatedData
+     * To create a new post of the user.
      */
-    public function createPost(User $user, array $validatedData): UserPost;
+    public function createPost(User $user, array $validatedData): UserPost; // @phpstan-ignore-line
```
Comment (bpocrafael) [inline laravel-app/app/Interfaces/PostServiceInterface.php]: remove @phpstan-ginore-line

**Review by bpocrafael — CHANGES_REQUESTED** [review]

File: laravel-app/app/Services/PostService.php (hunk 2 above)
Comment (adrean-hipe) [reply laravel-app/app/Services/PostService.php]: sir the one you commented is the outdated version of the PR, I already removed them from last 2 commits. Thank you

File: laravel-app/app/Services/PostService.php (hunk 3 above)
Comment (adrean-hipe) [reply laravel-app/app/Services/PostService.php]: sir the one you commented is the outdated version of the PR, I already removed them from last 2 commits. Thank you

File: laravel-app/app/Interfaces/PostServiceInterface.php (hunk 4 above)
Comment (adrean-hipe) [reply laravel-app/app/Interfaces/PostServiceInterface.php]: sir the one you commented is the outdated version of the PR, I already removed them from last 2 commits. Thank you

File: laravel-app/database/migrations/2023_10_09_030758_add_original_post_id_to_user_posts_table.php (hunk 1 above)
Comment (adrean-hipe) [reply laravel-app/database/migrations/2023_10_09_030758_add_original_post_id_to_user_posts_table.php:12]: done sir, added reverse. Thank you ❤️

File: laravel-app/phpstan.neon [hunk 5]
```diff
@@ -14,4 +16,4 @@ parameters:
     checkGenericClassInNonGenericObjectType: false
     checkUnusedViews: true
     excludePaths:
-        - bootstrap/cache
+        - bootstrap/cache
```
Comment (bpocrafael) [inline laravel-app/phpstan.neon:19]: add newline

**Review by bpocrafael — DISMISSED** [review]

File: laravel-app/phpstan.neon [hunk 6]
```diff
@@ -3,6 +3,8 @@ includes:
     - ./vendor/nunomaduro/larastan/extension.neon
 parameters:
     reportUnmatchedIgnoredErrors: false
+    ignoreErrors:
+        - '#with no value type specified in iterable type array\.$#'
```
Comment (bpocrafael) [inline laravel-app/phpstan.neon:7]: remove this one.

Comment (adrean-hipe) [reply laravel-app/phpstan.neon]: removed ignoreErrors sir. Thank you ❤️

File: laravel-app/phpstan.neon (hunk 5 above)
Comment (adrean-hipe) [reply laravel-app/phpstan.neon]: added new line sir, thank you ❤️

**Review by bpocrafael — APPROVED** [review]
🐑

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**